XamarinCommunityToolkit.UnitTests/Behaviors/UserStoppedTypingBehavior_Tests.cs

```diff
@@ -68,13 +69,84 @@ public async Task KeyboardDimissesAfterTimeThresholdExpires()
 			entry.Focus();
 			entry.Text = "1";
 
-			await Task.Delay(defaultThreshold + 100);
+			await Task.Delay(defaultTimeThreshold + 100);
 
 			// assert
 			Assert.False(entry.IsFocused);
 		}
 
-		public Entry CreateEntryWithBehavior(int threshold = defaultThreshold,
+		[Fact]
+		public async Task ShouldExecuteCommandWhenMinimumLengthThreholdHasBeenReached()
+		{
+			// arrange
+			var commandHasBeenExecuted = false;
+			var entry = CreateEntryWithBehavior(command: new Command<string>((s) => commandHasBeenExecuted = true),
+												lengthThreshold: 3);
+
+			// act
+			entry.Text = "1";
+			entry.Text = "12";
+			entry.Text = "123";
+			await Task.Delay(defaultTimeThreshold + 100);
+
+			// assert
+			Assert.True(commandHasBeenExecuted);
+		}
+
+		[Fact]
+		public async Task ShouldNotExecuteCommandWhenMinimumLengthThreholdHasNotBeenReached()
+		{
+			// arrange
+			var commandHasBeenExecuted = false;
+			var entry = CreateEntryWithBehavior(command: new Command<string>((s) => commandHasBeenExecuted = true),
+												lengthThreshold: 2);
+
+			// act
+			entry.Text = "1";
+			await Task.Delay(defaultTimeThreshold + 100);
+
+			// assert
+			Assert.False(commandHasBeenExecuted);
+		}
+
+		/// <summary>
+		/// Due to Focus() not setting the Entry to IsFocused = true, we cannot test if the entry still got focus or not
+		/// See for more information: https://forums.xamarin.com/discussion/181096/how-to-focus-an-entry-control-in-a-unit-test
+		/// </summary>
+		/// <returns></returns>
+		//[Fact]
+		//public async Task ShouldNotDismissKeyboardWhenMinimumLengthThreholdHasNotBeenReached()
+		//{
+		//	// arrange
+		//	var entry = CreateEntryWithBehavior(lengthThreshold: 3,
+		//										shouldDismissKeyboardAutomatically: true);
+
+		//	// act
+		//	entry.Focus();
```

Comment:
Can we try to work around it anyhow?

```csharp
entry.SetValue(Entry.IsFocusedProperty, true);
```

or

Invoking it in MainThread

Reply:
Hi @AndreiMisiukevich,

I'd already tried to invoke it on the MainThread, but then I'm getting the following error. And I don't know 100% sure, but you can only call the Xamarin.Forms.Init() only when running on a specific platform, which is not possible in a unit test, right (or at least not that easily)?

`System.InvalidOperationException : You MUST call Xamarin.Forms.Init(); prior to using it.`

When calling the SetValue on the entry field, I'm getting this error. Which makes sense as well.

`The BindableProperty "IsFocused" is readonly.`

Reply:
Let's remove this commented code then